compiler/daml-lf-verify/src/DA/Daml/LF/Verify/Context.hs

```diff
@@ -378,7 +587,7 @@ fresh :: MonadEnv m ph => m String
 fresh = do
   (cur,env) <- get
   put (cur + 1,env)
-  return $ show cur
+  return $ "_" ++ show cur
```

Comment:
Does this solve any issue or are you just trying to get nicer names? Variable names in DAML-LF can start with `_` so I don’t believe this helps with ensuring that the variable is fresh. In fact it seems to make it worse since they cannot start with a digit.

Reply:
This actually solves an issue! :) 
Every variable gets refreshed at least once, by adding `_Number` to it.
- Variables ending on an underscore don't pose any issue. e.g. if you have both variable `x` and `x_`, they would be translated to `x_Number` and `x__Number` which are different.
- Without the underscore, there is an issue though. Consider having variables `x1` and `x`. While refreshing them, we would add a unique number to them. If those numbers happened to be `2` and `12` for example, the refreshed variables would suddenly collide as `x12`. However, with the underscores this would result in `x1_2` and `x_12`.